pilot/model/config.go

```diff
@@ -497,7 +533,66 @@ func (store *istioConfigStore) RouteRules(instances []*ServiceInstance, destinat
 	return out
 }
 
+func (store *istioConfigStore) routeRulesV2(instances []*ServiceInstance, destination string) []Config {
+	out := make([]Config, 0)
+	configs, err := store.List(V1alpha2RouteRule.Type, NamespaceAll)
+	if err != nil {
+		return nil
+	}
+
+	for _, config := range configs {
+		rule := config.Spec.(*routingv2.RouteRule)
+
+		var found bool
+		for _, host := range rule.Hosts {
+			fqdn := ResolveFQDN(config.ConfigMeta, host)
+			if fqdn == destination {
+				found = true
+				break
+			}
+		}
+		if !found {
+			continue
+		}
+
+		if !matchSource(rule, instances) {
+			continue
+		}
```

Comment:
Not sure if this applies here anymore.. We have an array of rules. So we probably have to do the source match when iterating over individual rules.